packages/core/src/tracing/spanBuffer.js

```diff
@@ -147,14 +136,30 @@ exports.activate = function activate(extraConfig) {
   //       the AWS runtime might execute the handler in a different lambda execution.
   //       On AWS Lambda we wait till the handler finishes and then transmit all collected spans via
   //       `sendBundle`. Any detected span will be sent directly to the BE.
+  // TODO: This is not a good approach, because it assumes that the agent is ready.
+  //       Spans are collected during the agent cycle -  we flush them here and assume we
+  //       are connected to the agent.
   if (!isFaaS) {
-    transmissionTimeoutHandle = setTimeout(transmitSpans, initialDelayBeforeSendingSpans);
+    transmissionTimeoutHandle = setTimeout(transmitSpans, transmissionDelay);
     transmissionTimeoutHandle.unref();
   }
 
   if (preActivationCleanupIntervalHandle) {
     clearInterval(preActivationCleanupIntervalHandle);
   }
+
+  if (!isFaaS) {
```

Comment:
Not for Faas (currently only Lambda) because they send a bundle request at the end of the execution.